### PR TITLE
Adds Lab 4.

### PR DESCRIPTION
**Pull request by bvshih**
@AliSalman41

### CODE REVIEW

**Review by AliSalman41 — COMMENTED** [review]
problem 1: what's the point of the percent error plot? Go into it a bit. Also, you had some code that didn't translate into anything above cell 138. Explain why your initial model did not work that well and how it can be improved, go through the whole process step by step because I am a little confused as to why you needed to find the mode in cell 113. In cell 9, perhaps explain why you used the mod operator?

Problem 2: there is barely an explanation here so it is hard to follow. I understand that you are trying to build your background distribution but to someone who has not done this lab, they would not know what's going on. What do you get from the images shown in cell 34, I can see differences between images but you should explain what this means conceptually. Same thing with the images in the end, what are your findings?

**Comment by bvshih** [comment]
@hrzhao76
